libsolidity/interface/OptimiserSettings.h

```diff
@@ -111,27 +111,8 @@ struct OptimiserSettings
 		util::unreachable();
 	}
 
-	bool operator==(OptimiserSettings const& _other) const
-	{
-		return
-			runOrderLiterals == _other.runOrderLiterals &&
-			runInliner == _other.runInliner &&
-			runJumpdestRemover == _other.runJumpdestRemover &&
-			runPeephole == _other.runPeephole &&
-			runDeduplicate == _other.runDeduplicate &&
-			runCSE == _other.runCSE &&
-			runConstantOptimiser == _other.runConstantOptimiser &&
-			simpleCounterForLoopUncheckedIncrement == _other.simpleCounterForLoopUncheckedIncrement &&
-			optimizeStackAllocation == _other.optimizeStackAllocation &&
-			runYulOptimiser == _other.runYulOptimiser &&
-			yulOptimiserSteps == _other.yulOptimiserSteps &&
-			expectedExecutionsPerDeployment == _other.expectedExecutionsPerDeployment;
-	}
-
-	bool operator!=(OptimiserSettings const& _other) const
-	{
-		return !(*this == _other);
-	}
+	bool operator==(OptimiserSettings const& _other) const = default;
+	bool operator!=(OptimiserSettings const& _other) const = default;
```

Comment:
I thought this was a good opportunity to use C++20 default comparison operators now that we have them.  While changing this, I noticed that `yulOptimiserCleanupSteps` were not part of the comparison. I guess this was not intentional. In any case, we can also do this in another PR if you prefer.

Reply:
Oh, that's pretty bad. Definitely a separate PR.

We use the operator to check if the current optimizer settings match defaults. This means that we'll ignore this field when we generate metadata and all other fields are set to defaults: https://github.com/ethereum/solidity/blob/e0e661f6ba514bcf692d8cf15f3bf72678beb69a/libsolidity/interface/CompilerStack.cpp#L1752-L1755

This is a bug similar to #15641, but a little more serious because the setting is a bit more likely to be used. If I'm right, this will result in bytecode not reproducible from metadata.

Reply:
Oops, yeah, looks like it. -> PR #15859

Reply:
I have dropped the commit with this change from this PR to reduce noise.